tests/stubs.py

```diff
@@ -26,6 +30,26 @@ def apdu(*args):
     return {"data": EndPoint.APDU + status}
 def button(*args):
     return {}
+iter = 0
+idx = 1
+def screenshot(*args):
+    global iter
+    global idx
+    path = Path(__file__).parent.resolve() / "snapshots/nanos/generic" / f"{str(idx).zfill(5)}.png" 
+    sleep(0.25)
+    if iter==1:
```

Comment:
@lpascal-ledger ugly but I don't really know how to stub it in a smarter way.

Reply:
Maybe with a class? They are useful as local namespaces:
```python
class Screenshots:
    def __init__(self):
        self.iter = 0
        self.idx = 1
        self.steps = {1: 0, 5: 1, 8: 2, 13: 3}

    def fetch(self):
        path = Path(__file__).parent.resolve() / "snapshots/nanos/generic" / f"{str(self.idx).zfill(5)}.png"
        self.idx = self.steps.get(self.iter, self.idx)
        self.iter += 1
        img_temp = Image.open(path)
        iobytes = BytesIO()
        img_temp.save(iobytes, format="PNG")
        return iobytes.getvalue(),200
[...]

class SpeculosServerStub:
    def __init__(self):
        screenshots = Screenshots()
        [...]
        self.app.add_url_rule("/screenshot", methods=["GET"], view_func=screenshots.fetch)
```

(Was the `sleep` necessary?)

Reply:
@lpascal-ledger I updated the archi based on your advice and with a smarter approach (use button requests to update index of image, of course !)

`sleep` was necessary with first implem. I removed it.